### PR TITLE
Fixed instanceof summary to better match instanceof page

### PR DESCRIPTION
**Pull request by LostBeard**
### Description

The wording describing [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_operators#instanceof) on the `Expressions and operators` page does not match the wording on the page for [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof) and could lead to a misunderstanding of how it functions.

From:
[Expressions and operators - instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Expressions_and_operators#instanceof)
Current wording:
> ```js-nolint
> objectName instanceof objectType
> ```
>
> where `objectName` is the name of the object to compare to `objectType`, and `objectType` is an object type, such as {{jsxref("Date")}} or {{jsxref("Array")}}.

Updated (in patch):
> ```js-nolint
> object instanceof objectType
> ```
>
>where `object` is the object to test against `objectType`, and `objectType` is an object type, such as {{jsxref("Date")}} or {{jsxref("Array")}}.

The updated wording matches the wording on the [instanceof](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/instanceof) page.

### Motivation

I found the issue while using MDN and I feel new users may get confused by the current wording and example.

### CODE REVIEW

**Review by Josh-Cena — COMMENTED** [review]
Could you explain how they are different before and after? They mean the same thing to me.

**Comment by LostBeard** [comment]
@Josh-Cena 
> Could you explain how they are different before and after? They mean the same thing to me.

The after directly matches the wording on the page I also linked to.

As used in the example I quoted, `objectName` is used to represent an object.

An object name is a string used to **represent** an object in code. An object name and an object are not the same thing. 

I have no problem with you leaving the site inconsistent and semantically incorrect. 🤷‍♂️

**Comment by Josh-Cena** [comment]
In syntax examples, `objectName` is not a string. `objectName` is an identifier referring to an object. For example if I say `{ propertyName: propertyValue }` you wouldn't think that `propertyName` should be replaced with a string literal; you understand that it's a syntax which means `propertyName` can be replaced with any syntactically valid property name. I don't think the current content is incorrect. "Inconsistent", sure, but syntax examples are not something we are interested in being consistent on because it's unenforceable.

> `objectName` was intentionally used instead of `object` to not confuse people. `object` is too similar to `Object`.

@estelle Unfortunately it's not _that_ intentional here either, as we often distinguish between lowercase and uppercase values. Both are applicable, it's just that there's no strong incentive to change one for the other.

However, I do agree that the guide should try to be less confusion-prone than reference, since the reference is for more advanced readers that are likely already familiar with the syntax.

**Comment by Josh-Cena** [comment]
Actually never mind, let's do this. `objectName` is a terrible name since it implies only "names" can be used, but in fact you can use any expression here such as `foo.bar instanceof Type`.

File: files/en-us/web/javascript/guide/expressions_and_operators/index.md [hunk 1]
```diff
@@ -1035,10 +1035,10 @@ The [`instanceof` operator](/en-US/docs/Web/JavaScript/Reference/Operators/insta
 if the specified object is of the specified object type. The syntax is:
 
 ```js-nolint
-objectName instanceof objectType
+object instanceof objectType
 ```
 
-where `objectName` is the name of the object to compare to `objectType`, and `objectType` is an object type, such as {{jsxref("Date")}} or {{jsxref("Array")}}.
+where `object` is the object to test against `objectType`, and `objectType` is an object type, such as {{jsxref("Date")}} or {{jsxref("Array")}}.
```
Comment (Josh-Cena) [inline files/en-us/web/javascript/guide/expressions_and_operators/index.md:1041]:
```suggestion
where `object` is the object to test against `objectType`, and `objectType` is a constructor, such as {{jsxref("Date")}} or {{jsxref("Array")}}.
```

**Review by Josh-Cena — APPROVED** [review]
Thanks @LostBeard! I'm now convinced this is an improvement.

**Pull request merged**